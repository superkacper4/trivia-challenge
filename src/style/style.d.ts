import 'styled-components';

// and extend it
declare module 'styled-components' {
    export interface DefaultTheme {
        fontSizes: {
            desktop: {
                h1: string;
                m: string;
                button: string;
                question: string;
                currentQuestion: string;
                questionOverview: string;
                level: string;
                topic: string;

            },
            mobile: {
                h1: string;
                m: string;
                button: string;
                question: string;
                currentQuestion: string;
                questionOverview: string;
                level: string;
                topic: string;
            }
        }

        colors: {
            main: string;
            secondary: string;
            third: string;
            fourth: string;
            fivth: string;
            sixth: string;
            seventh: string;
            eighth: string;
            white: string;
        };

        breakPoints: {
            phone: string;
            tablet: string;
            desktop: string;
            bigDesktop: string;
        }
    }
}