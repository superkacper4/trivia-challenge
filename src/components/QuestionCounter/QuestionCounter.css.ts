import styled from 'styled-components'

interface types {
    score: boolean;
}

const StyledP = styled.p`
    width: ${({ score }: types) => score ? 'auto' : '100%'};
    margin-left: ${({ score }: types) => score ? '10px' : 'none'};
    font-size: ${({ theme }) => theme.fontSizes.mobile.level};
`;

const StyledSpan = styled.span`
    font-size: ${({ theme }) => theme.fontSizes.mobile.currentQuestion};
    color: ${({ theme }) => theme.colors.third};
`;

export { StyledP, StyledSpan }