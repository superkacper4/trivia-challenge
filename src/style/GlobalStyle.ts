import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
    body{
        background-color: ${({ theme }) => theme.colors.main};
        overflow-x: hidden;
        padding: 0;
        margin: 0;
        font-family: 'Quicksand', sans-serif;
        /* height: 100vh; */
    }
    * {
        box-sizing: border-box;
        -webkit-font-smoothing: antialiased;
        -moz-osx-font-smoothing: grayscale;
    }
`;

export default GlobalStyle;
