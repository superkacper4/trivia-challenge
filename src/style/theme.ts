import { DefaultTheme } from 'styled-components';

const theme: DefaultTheme = {
    fontSizes: {
        desktop: {
            h1: '46px',
            m: '18px',
            button: '21px',
            question: '31px',
            questionOverview: '16px',
            level: '28px',
            topic: '81px',

        },
        mobile: {
            h1: '26px',
            m: '13px',
            button: '15px',
            question: '25px',
            questionOverview: '12px',
            level: '13px',
            topic: '31px',
        }
    },

    colors: {
        main: '#424A9E',
        secondary: '#E5E5E5',
        third: '#FFA67A',
        fourth: '#FF6065',
        fivth: '#FF6065',
        white: 'white',
    }
};

export default theme;