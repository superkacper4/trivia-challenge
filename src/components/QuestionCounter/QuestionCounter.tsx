import React from 'react'
import { StyledP, StyledSpan } from './QuestionCounter.css'

interface types {
    currentQuestion: number;
    amount: number;
    score: boolean;
}


const QuestionCounter = ({ currentQuestion, amount, score }: types) => (
    <StyledP score={score}>
        <StyledSpan>{currentQuestion < 10 ? `0${currentQuestion}` : currentQuestion}</StyledSpan>/{amount}
    </StyledP>
)

export default QuestionCounter