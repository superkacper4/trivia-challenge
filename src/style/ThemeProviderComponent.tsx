import React, { ReactNode } from 'react'
import { ThemeProvider } from 'styled-components'
import theme from './theme'

interface Types {
    children: ReactNode
}

const ThemeProviderComponent = ({ children }: Types) => (
    <ThemeProvider theme={theme}>
        {children}
    </ThemeProvider>
)

export default ThemeProviderComponent