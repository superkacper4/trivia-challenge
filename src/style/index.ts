import ThemeProviderComponent from './ThemeProviderComponent'
import GlobalStyle from './GlobalStyle'
import theme from './theme'

export { GlobalStyle, ThemeProviderComponent, theme }