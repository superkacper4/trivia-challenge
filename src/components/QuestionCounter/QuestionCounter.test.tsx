import { render, screen } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../style';
import QuestionCounter from './QuestionCounter';

describe('QuestionCoutner', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <QuestionCounter currentQuestion={2} amount={10} score />
      </ThemeProvider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render 0 if amount is lower than 10', () => {
    render(
      <ThemeProvider theme={theme}>
        <QuestionCounter currentQuestion={2} amount={9} score />
      </ThemeProvider>,
    );
    expect(screen.getByTestId('question-counter-span')).toHaveTextContent('02');
  });

  it('should not render 0 if amout is higher/equal than 10', () => {
    render(
      <ThemeProvider theme={theme}>
        <QuestionCounter currentQuestion={10} amount={11} score />
      </ThemeProvider>,
    );
    expect(screen.getByTestId('question-counter-span')).toHaveTextContent('10');
  });
});
