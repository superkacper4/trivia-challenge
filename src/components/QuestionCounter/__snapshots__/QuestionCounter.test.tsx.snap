// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionCoutner should match the snapshot 1`] = `
<DocumentFragment>
  <p
    class="sc-bdnxRM kEYANI"
  >
    <span
      class="sc-gtsrHT fqcFIy"
      data-testid="question-counter-span"
    >
      02
    </span>
    /10
  </p>
</DocumentFragment>
`;
